### PR TITLE
Get base code working

### PR DESCRIPTION
**Pull request by Thatoddone1**
This branch adds basic code for the drive train(taken from last year's code), the lifter, and claw(might have to lower speed depending on the design). It does not include the code for autonomous, or for the second motor on the lift mechanism. I am not sure if I did everything how you want it, but I tried. I also added in some of the VSCode Vex configuration.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
+#CONFIG
 
 brain=Brain()
 
 controller=Controller(PRIMARY)
 
 
+
+
```
Comment (ghost) [inline src/main.py]: Can you reduce the blank lines throughout the repo? At most should be 2-3 blank lines, no more as harder to navigate

File: src/main.py [hunk 2]
```diff
@@ -20,20 +26,65 @@ def on_down_button_pressed():
...
-controller.buttonR1.released(on_button_released)
+controller.buttonR1.released(on_lift_button_released)
 controller.buttonR2.pressed(on_down_button_pressed)
-controller.buttonR2.released(on_button_released)
+controller.buttonR2.released(on_lift_button_released)
+
+
+####### CLAW CODE ↓ ########
+
+claw = Motor(Ports.PORT12)
```
Comment (ghost) [inline src/main.py:31]: Do you think it would be easier to move the declarations of the motors to the top?

File: src/main.py [hunk 3]
```diff
@@ -20,20 +26,65 @@ def on_down_button_pressed():
...
+
+
+
+####### DRIVE STUFF ↓ ########
+
+
+
+# motors and drive train setup
+left_motor_a = Motor(Ports.PORT1, GearSetting.RATIO_18_1, False)
+left_motor_b = Motor(Ports.PORT2, GearSetting.RATIO_18_1, False)
```
Comment (ghost) [inline src/main.py:53]: Also potentially move to top

File: src/main.py [hunk 4]
```diff
@@ -45,8 +96,8 @@ def rc_auto_loop_function_controller_1():
             # calculate the drivetrain motor velocities from the controller joystick axies
             # left = axis3 + axis1
             # right = axis3 - axis1
-            drivetrain_left_side_speed = controller_1.axis3.position()
-            drivetrain_right_side_speed = controller_1.axis2.position()
+            drivetrain_left_side_speed = controller.axis3.position()
```
Comment (ghost) [inline src/main.py:75]: If you rename from controller_1 to controller, I would also recommend renaming all variables that contain controller_1

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main.py (hunk 2 above)
Comment (Thatoddone1) [reply src/main.py:31]: I actually think that it makes more sense to keep them with the rest of the code for that part. Keeping everything related to the claw with the claw I think makes it easier to keep stuff organized. But it is up to you.

File: src/main.py [hunk 5]
```diff
@@ -20,20 +26,65 @@ def on_down_button_pressed():
...
+
+#varibles needed for the controoler loop
 drivetrain_l_needs_to_be_stopped_controller_1 = False
 drivetrain_r_needs_to_be_stopped_controller_1 = False
 
+
+
 # define a task that will handle monitoring inputs from controller_1
 def rc_auto_loop_function_controller_1():
     global drivetrain_l_needs_to_be_stopped_controller_1, drivetrain_r_needs_to_be_stopped_controller_1, controller_1_left_shoulder_control_motors_stopped, controller_1_x_b_buttons_control_motors_stopped, remote_control_code_enabled
```
Comment (ghost) [inline src/main.py]: none of these variables are being used

File: src/main.py [hunk 6]
```diff
@@ -20,20 +26,65 @@ def on_down_button_pressed():
...
+left_motor_a = Motor(Ports.PORT1, GearSetting.RATIO_18_1, False)
+left_motor_b = Motor(Ports.PORT2, GearSetting.RATIO_18_1, False)
+left_drive_smart = MotorGroup(left_motor_a, left_motor_b)
+right_motor_a = Motor(Ports.PORT3, GearSetting.RATIO_18_1, True)
+right_motor_b = Motor(Ports.PORT4, GearSetting.RATIO_18_1, True)
+right_drive_smart = MotorGroup(right_motor_a, right_motor_b)
+drivetrain = DriveTrain(left_drive_smart, right_drive_smart, 319.19, 355.59999999999997, 266.7, MM, 1)
+
+#varibles needed for the controoler loop
 drivetrain_l_needs_to_be_stopped_controller_1 = False
```
Comment (ghost) [inline src/main.py]: Delete these also

File: src/main.py [hunk 7]
```diff
@@ -1,42 +1,70 @@
...
+####### CLAW CODE ↓ ########
 
-# define a task that will handle monitoring inputs from controller_1
-def rc_auto_loop_function_controller_1():
-    global drivetrain_l_needs_to_be_stopped_controller_1, drivetrain_r_needs_to_be_stopped_controller_1, controller_1_left_shoulder_control_motors_stopped, controller_1_x_b_buttons_control_motors_stopped, remote_control_code_enabled
+claw = Motor(Ports.PORT12)
+
+claw.set_stopping(BrakeType.HOLD)
+
+
```
Comment (ghost) [inline src/main.py]: Remove space on line 34 to be consistent

**Review by ghost — APPROVED** [review]
Just fix that one comment and you can merge

**Review by ghost — APPROVED** [review]

**Pull request merged**